website/profile/views.py

```diff
@@ -403,9 +402,8 @@ def oauth_application_detail(auth, **kwargs):
 
     # The client ID must be an active and existing record, and the logged-in user must have permission to view it.
     try:
-        #
-        record = ApiOAuth2Application.find_one(Q('client_id', 'eq', client_id))
-    except NoResultsFound:
+        record = ApiOAuth2Application.objects.get(client_id=client_id)
```

Comment:
I actually tried using it here (and in the other places you've mentioned), but `get_object_or_404` returns an [`Http404`](https://docs.djangoproject.com/en/1.11/topics/http/views/#django.http.Http404) whereas our v1 views are expecting our custom [`HTTPError`](https://github.com/CenterForOpenScience/osf.io/blob/develop/framework/exceptions/__init__.py#L11).

The tl;dr of it is that this works just fine in APIv2, but returning `Http404` broke our beautiful "Page not found" templating in [`error.mako`](https://github.com/CenterForOpenScience/osf.io/blob/develop/website/templates/error.mako). While the shortcut would be nice, it seemed out of scope to change the error handling right now.